### PR TITLE
Set up associations

### PR DESCRIPTION
**Pull request by Yothu**
# Description

Add associations to some of the project's classes, and add some classes.

## Requirements
- [x] Create a `Classroom` class.
  - [x] Associate the `Classroom` class and the `Student` class so that a classroom **has many** students, and a student **belongs to** a classroom.
  - [x] Add a method to the `Classroom` class so that it adds a student to the class.
  - [x] Add a method to the `Student` class so that it sets a classroom for the student.
- [x] Create a `Book` class.
- [x] Create a `Rental` class.
  - [x] Associate the `Person` class and the `Book` class so that they have a **many to many** relationship using the `Rental` class as an intermediate class.
- [x] Each class should be in a separate file.

## References
See this [page](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_associations.md) to know the instructions for this project in more detail!
See the [README.md](https://github.com/Yothu/oop-school-library/blob/feature/set-up-associations/README.md) and check the installation and usage.

### CODE REVIEW

File: rental.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Rental class
+class Rental
+  attr_accessor :date
+  attr_reader :person, :book
```
Comment (TedLivist) [inline rental.rb]: - [x] The `book` and `person` variables here should have getter and setter (`attr_accessor`), not just getter (`attr_reader`). Kindly refactor accordingly.

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# Book class
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
```
Comment (TedLivist) [inline book.rb]: - [x] Kindly set the getter and setter for `:rentals` here instead of getter only.

File: person.rb [hunk 3]
```diff
@@ -2,7 +2,7 @@
 
 # Person Class
 class Person < Nameable
-  attr_reader :id
+  attr_reader :id, :rentals
   attr_accessor :name, :age, :parent_permission
```
Comment (TedLivist) [inline person.rb]: - [x] Getter and setter for `:rentals` should be present, not just getter. Kindly refactor accordingly.

File: book.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+end
```
Comment (TedLivist) [inline book.rb]: - [x] You are missing the method for adding rentals. Kindly do that.

  You can import the **Rental** class here. Then set up an `add_rental` method that takes in `person` and `date` parameters and return an instance of the Rental object like so:
  ```
  def add_rental(date, person)
    Rental.new(date, person, self)     #=> notice the order of parameters
  end
  ```

File: person.rb [hunk 5]
```diff
@@ -2,7 +2,7 @@
 
 # Person Class
 class Person < Nameable
```
Comment (TedLivist) [inline person.rb:5]: - [x] Kindly add the method for adding rentals. You can take the approach suggested earlier: Import Rental class and note the order of the parameters

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @Yothu,

Good job so far!
✔️ Set up of associations
✔️ Good documentation
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: rental.rb (hunk 1 above)
Comment (Yothu) [reply rental.rb]: The `date` variable already has `attr_accessor` though.

Comment (TedLivist) [reply rental.rb]: @Yothu sorry about that. Thanks for pointing it out. It was intended the other way. `:person` and `:book` should have getter and setter instead, not just getter

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: INVALID CODE REVIEW REQUEST 🛑 🛑

Hi @Yothu,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

### You have submitted a Code Review Request for Add Basic UI, Whereas you are working on the Set up Associations project. Kindly cross-check your form before submitting.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: rental.rb (hunk 1 above)
Comment (rloterh) [reply rental.rb]: This issue is well resolved! ✔️

File: book.rb (hunk 2 above)
Comment (rloterh) [reply book.rb]: This issue is well resolved! ✔️

File: person.rb (hunk 3 above)
Comment (rloterh) [reply person.rb]: This issue is well resolved! ✔️

File: book.rb (hunk 4 above)
Comment (rloterh) [reply book.rb]: This issue is well resolved! ✔️

File: person.rb (hunk 5 above)
Comment (rloterh) [reply person.rb:5]: This issue is well resolved! ✔️

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Davih, 

Your project is complete! Great job for making the necessary changes recommended in the previous review. There is nothing else to say other than... it's time to merge it 🍾 :shipit:. Congratulations! 🎉🥂


<br>
Cheers and Happy coding!👏 💻 🍷 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 1 more limited review per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**